src/pages/events/index.svelte

```diff
@@ -1,90 +1,164 @@
-import { join, events, loading } from 'data/events.js'
-import { me } from 'data/user.js'
-
-import Icon from 'fa-svelte'
-import { faExternalLinkAlt, faEdit } from '@fortawesome/free-solid-svg-icons'
-import { goto, params } from '@roxi/routify'
-import { flip } from 'svelte/animate';
-import { fade } from 'svelte/transition';
-
-const format = date => new Date(date).toLocaleDateString([], {
-  month: 'short',
-  day: 'numeric',
-  year: 'numeric',
-  timeZone: 'UTC',
-})
-const logoUrl = event => event.image_file?.url || ''
-
-let eventFilter = {}
-
-$: eventFilter = populateEventFilterFrom($params, $me)
+  import EventFilters from "components/EventFilters.svelte";
+  import FilteredDisplay from "components/FilteredDisplay.svelte";
+  import {
+    isParticipant,
+    populateEventFilterFrom,
+  } from "data/eventFiltering.js";
+  import { join, events, loading } from "data/events.js";
+  import { me } from "data/user.js";
+
+  import Icon from "fa-svelte";
+  import { faExternalLinkAlt, faEdit } from "@fortawesome/free-solid-svg-icons";
+  import { goto, params } from "@roxi/routify";
+  import * as animateScroll from "svelte-scrollto";
+  import { flip } from "svelte/animate";
+  import { fade } from "svelte/transition";
+
+  const format = (date) =>
+    new Date(date).toLocaleDateString([], {
+      month: "short",
+      day: "numeric",
+      year: "numeric",
+      timeZone: "UTC",
+    });
+  const logoUrl = (event) => event.image_file?.url || "";
+
+  let eventFilter = {};
+  // let scrollTo = $params.scrollTo
+
+  $: eventFilter = populateEventFilterFrom($params, $me);
+
+  function scrollToEvent(filteredEvents) {
+    if ($params.scrollTo) {
+      let eventNum;
+
+      filteredEvents.forEach((val, i) => {
+        if (val.id == $params.scrollTo) eventNum = i + 1;
+      });
+
+      if (!eventNum) return;
+
+      let eventId = `#event-${eventNum}`;
+
+      setTimeout(() => {
+        animateScroll.scrollTo({
+          element: eventId,
+          duration: 1000,
+        });
+      }, 1000);
+
+      // if the 'scrollTo' url parameter should be removed from the url, uncomment this and the 'scrollTo'
+      // variable above and then change all the '$params.scrollTo' to just 'scrollTo'
+
+      // let stateObj = { id: "100" };
+      // window.history.replaceState(stateObj, "Events", "/events");
+
+    }
+
+    return "";
+  }
 </script>
 
 <style>
-.event-buttons-container .btn {
-  width: 10em;
-}
-
-.event-buttons-container .btn:disabled {
-  cursor: unset;
-}
-
-.logo {
-  display: flex;
-  align-items: center;
-  justify-content: center;
-}
-
-/* these will get used only on XS screens */
-.logo img {
-  max-height: 5rem;
-  margin-bottom: 2rem;
-}
-li {
-  max-height: initial;
-}
-
-/* https://getbootstrap.com/docs/4.4/layout/overview/#responsive-breakpoints */
-/* these will get used on everything above XS screens using bootstrap's breakpoint settings */
-@media (min-width: 576px) {
+  @keyframes event-blinking {
```

Comment:
I don't know much about animations, so I am not much help here. However, I believe Svelte has some animation capability as well. Did you check that out?

Reply:
I didn't. I'll look into this tomorrow.

Reply:
I did some research and I didn't really find anything with svelte that would do the same thing in fewer lines. The svelte animations deal more with position; I'm just wanting to change the border.